pkg/apis/core/validation/validation.go

```diff
@@ -6135,6 +6227,39 @@ func validateWindowsHostProcessPod(podSpec *core.PodSpec, fieldPath *field.Path,
 	return allErrs
 }
 
+// validateOS validates the OS field within pod spec
+func validateOS(podSpec *core.PodSpec, fldPath *field.Path, opts PodValidationOptions) field.ErrorList {
+	allErrs := field.ErrorList{}
+	// These valid values currently are linux, windows.
+	// In future, they can be expanded to values from
+	// https://github.com/opencontainers/runtime-spec/blob/master/config.md#platform-specific-configuration
+	validOS := sets.NewString(string(core.Linux), string(core.Windows))
+	os := podSpec.OS
+	if os == nil {
+		return allErrs
+	}
+	if len(os.Name) == 0 {
+		return allErrs
+	}
+	osName := strings.ToLower(string(os.Name))
```

Comment:
don't make this case-insensitive, require an exact match to the permitted values